os/kernel/task/task_setup.c

```diff
@@ -468,6 +468,11 @@ static int thread_schedsetup(FAR struct tcb_s *tcb, int priority, start_t start,
 		for (; i < 3 * MPU_NUM_REGIONS; i += 3)
 #endif
 		{
+#ifdef CONFIG_MPU_STACKGUARD
+			if (i == 9) {	/* skip already assigned stack protection */
```

Comment:
Why is this 9?

Reply:
It is the register address to be updated for MPU region number 3. 
Each MPU region will be enabled with changing minimum of 3 registers.
I have modified this as generic in my latest changes submitted